arrow-cast/src/cast/mod.rs

```diff
@@ -2888,96 +2770,103 @@ mod tests {
         );
     }
 
+    macro_rules! generate_decimal_to_numeric_cast_test_case {
+        ($INPUT_ARRAY: expr) => {
```

Comment:
```suggestion
    fn generate_decimal_to_numeric_cast_test_case<T>(array: &PrimitiveArray<T>)
    where
        T: ArrowPrimitiveType + DecimalType,
    {
```

I personally prefer generics, but not too much difference anyway